src/main/java/org/fenixedu/academic/domain/student/StatuteType.java

```diff
@@ -141,6 +142,10 @@ public void delete() {
         return readAll(StatuteType::isSpecialSeasonGrantedByRequest).findFirst();
     }
 
+    public static Optional<StatuteType> findExtraordinarySeasonGrantedStatuteType() {
+        return readAll(StatuteType::isExtraordinarySeasonGranted).findFirst();
```

Comment:
Would this be the best way to do this?
I didn't want to duplicate the "SpecialSeasonGrantedByRequest" since it was deprecated.
I feel like this might have unexpected behaviour if there was more than one statute with extraordinary season granted.

Reply:
I agree it's strange... and some code cleanup could avoid such solutions as findFirst or iterator.next...
In this case just to exactly that.
The purpose of the statutetype is to grant access to each evaluation season.
Some students indeed have multiple types that will grant them access, but which one we use has no real bearing on the result.